### DISCUSSION TITLE
An Inspector rework for all things GUI

### DISCUSSION THREAD

**Discussion by philippedcote** [godotengine/godot-proposals#4097] [GUI]
Coming from top GUI design tools like Figma, it’s a bit cumbersome to develop every possible Toggle Items in the Inspector whereas it could be much more visual and condensed, like this:

<img width="239" alt="image" src="https://user-images.githubusercontent.com/890553/155366436-7e8f4173-5b2f-4133-b7a8-42099d6048c6.png">
----
<img width="239" alt="image" src="https://user-images.githubusercontent.com/890553/155366708-78deeeb6-791d-4c0e-bb74-ac9bd9c68dac.png">

Using tabs could also help to separate "Advanced" settings from the rest.

**Comment by Calinou** [comment]
This was already reworked for 4.0 in https://github.com/godotengine/godot/pull/55157 (see the [original proposal](https://github.com/godotengine/godot-proposals/issues/2841)). It relies on backwards-incompatible changes, so it can't be backported to `3.x`.
